### PR TITLE
Use then-request Instead of sync-request

### PR DESCRIPTION
**Pull request by SudharakaP**
The library sync-request cannot be used in production applications (reference: https://www.npmjs.com/package/sync-request). In jhipster-online, sync-request is I think the root cause of generation being hanging. When I look at the processes always see a stuck process originating from this library. Therefore I would suggest replacing this with then-process instead.

Fixes https://github.com/jhipster/jhipster-online/issues/233

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mshima** [comment]
@SudharakaP I’ve adapted the api to async request. But did using mobile, no prettier available.
Probably this needs manual test.

File: generators/openapi-client/prompts.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ function askActionType() {
         this.props = props;
 
         if (props.jhipsterEndpoint !== undefined) {
-            return fetchSwaggerResources(props.jhipsterEndpoint).then(availableDocs => {
+            fetchSwaggerResources(props.jhipsterEndpoint).then(availableDocs => {
```
Comment (mshima) [inline generators/openapi-client/prompts.js:163]: This return is required.
```suggestion
            return fetchSwaggerResources(props.jhipsterEndpoint).then(availableDocs => {
```

File: generators/openapi-client/prompts.js [hunk 2]
```diff
@@ -152,31 +150,32 @@ function askActionType() {
...
+        if (props.jhipsterEndpoint !== undefined) {
+            fetchSwaggerResources(props.jhipsterEndpoint).then(availableDocs => {
+                props.availableDocs = availableDocs;
+            });
+        }
+        if (props.jhipsterDirectory !== undefined) {
+            props.inputSpec = `${props.jhipsterDirectory}/src/main/resources/swagger/api.yml`;
+        } else if (props.customEndpoint !== undefined) {
+            props.inputSpec = props.customEndpoint;
+        }
```
Comment (mshima) [inline generators/openapi-client/prompts.js:171]:
```suggestion
        }
        return undefined;
```

**Comment by SudharakaP** [comment]
@mshima : Thanks for modifying the openclient-api as well. I ran it on a project and it seems to work well without any errors. So I think we could merge this one if you are okay. 🤔

**Comment by mshima** [comment]
@SudharakaP fell free to merge. I didn’t test.
The test focus should be jhipsterEndpoint prompt validation.

**Comment by SudharakaP** [comment]
> @SudharakaP fell free to merge. I didn’t test.
> The test focus should be jhipsterEndpoint prompt validation.

@mshima : Yes it seems to work well. So I am merging. 😄

**Pull request merged**